### PR TITLE
slack-config: Add Jeremy Rickard as a SIG Release Technical Lead

### PR DESCRIPTION
**Pull request by hasheddan**
Adds @jeremyrickard to sig-release slack groups as part of technical lead
onboarding.

Signed-off-by: hasheddan <georgedanielmangum@gmail.com>

**Which issue(s) this PR fixes**:

xref https://github.com/kubernetes/sig-release/issues/1421

/hold
/assign @justaugustus @saschagrunert

### CODE REVIEW

File: communication/slack-config/sig-release/usergroups.yaml
Comment (justaugustus) [inline communication/slack-config/sig-release/usergroups.yaml:23]: Drop.

File: communication/slack-config/sig-release/usergroups.yaml
Comment (justaugustus) [inline communication/slack-config/sig-release/usergroups.yaml:99]: Drop.

**Review by justaugustus — CHANGES_REQUESTED** [review]
@hasheddan -- Roughly the same review ashttps://github.com/kubernetes/k8s.io/pull/1529#pullrequestreview-567793308.

There are a few of us in @kubernetes/sig-release-leads that hold multiple roles...
Yourself, @saschagrunert, and I are:

- SIG leadership
- Release Managers
- RelEng subproject owners

We need to codify this somewhere, but I've been roughly going by these rules:

- All SIG Release leadership should be in `sig-release-leads` and `release-team-leads`
- Only Release Managers and Release Manager Associates should be in `release-managers`
- Only _full-fledged_ members of both Release Managers and the Product Security Committee should be in `security-release-team`

/hold
/retitle slack-config: Add Jeremy Rickard as a SIG Release Technical Lead

**Comment by hasheddan** [comment]
@justaugustus updates made :+1:

**Comment by justaugustus** [comment]
Awesome, thanks @hasheddan!
/lgtm
/approve
/hold cancel

**Pull request merged**
